dronekit/__init__.py

```diff
@@ -1282,7 +1282,7 @@ def listener(_):
             if not self._params_start:
                 return
 
-            if None not in self._params_set and not self._params_loaded:
+            if not self._params_loaded and all(x is not None for x in self._params_set):
```

Comment:
Is this correct?  Is it actually faster?

Reply:
Micro-benchmark of `None not in` vs. `all(x is not None...)`:

```python
from pymavlink.dialects.v20.common import MAVLink_param_value_message

import timeit


def main():
    list_to_check = [
        MAVLink_param_value_message(param_id='param_{}'.format(i), param_value=1, param_type=4, param_count=501, param_index=i)
        for i in range(500)
    ] + [None]

    assert not all(x is not None for x in list_to_check)

    time_original = timeit.timeit("None not in list_to_check", number=100000, globals=locals())
    time_optimized = timeit.timeit("all(x is not None for x in list_to_check)", number=100000, globals=locals())

    print("Original time: {:.4f}; optimized time: {:.4f}".format(time_original, time_optimized))


if __name__ == '__main__':
    main()
```

Output:

    Original time: 7.4819; optimized time: 3.1870

Reply:
Once the parameter are loaded, though, the `and` short-circuiting kicks in: `not self._params_loaded` becomes True and the list does not even get checked.

Reply:
Well, yes, but I find the other change makes things less clear so it would need to be a beneficial change :-)

The reordering is good.